src/handlers/records-query.ts

```diff
@@ -144,21 +144,21 @@ export class RecordsQueryHandler implements MethodHandler {
   private async fetchRecordsAsNonOwner(
     tenant: string, recordsQuery: RecordsQuery
   ): Promise<{ messages: GenericMessage[], cursor?: PaginationCursor }> {
-    const { dateSort, pagination } = recordsQuery.message.descriptor;
+    const { dateSort, pagination, filter } = recordsQuery.message.descriptor;
     const filters = [];
-    if (RecordsQueryHandler.filterIncludesPublishedRecords(recordsQuery)) {
+    if (Records.filterIncludesPublishedRecords(filter)) {
       filters.push(RecordsQueryHandler.buildPublishedRecordsFilter(recordsQuery));
     }
 
-    if (RecordsQueryHandler.filterIncludesUnpublishedRecords(recordsQuery)) {
+    if (Records.filterIncludesUnpublishedRecords(filter)) {
       filters.push(RecordsQueryHandler.buildUnpublishedRecordsByQueryAuthorFilter(recordsQuery));
 
       const recipientFilter = recordsQuery.message.descriptor.filter.recipient;
       if (recipientFilter === undefined || recipientFilter === recordsQuery.author) {
         filters.push(RecordsQueryHandler.buildUnpublishedRecordsForQueryAuthorFilter(recordsQuery));
       }
 
-      if (RecordsQueryHandler.shouldProtocolAuthorizeQuery(recordsQuery)) {
+      if (Records.hasProtocolRole(recordsQuery.signaturePayload!)) {
```

Comment:
Note to self: let's discuss this, it seems the original name describes the intent better, but want to hear what you are thinking.

Reply:
renamed to `shouldProtocolAuthorize`. 

My original thinking was that the method doesn't do much, so maybe it should describe what it is actually doing, but after further consideration I renamed it and the comment really helps the user see what's going on without clicking into the function.